pkgs/by-name/co/codecrafters-cli/package.nix

```diff
@@ -0,0 +1,38 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, git
+}:
+
+buildGoModule rec {
+  pname = "codecrafters-cli";
+  version = "34";
+  commit = "38dc748";
+
+  src = fetchFromGitHub {
+    owner = "codecrafters-io";
+    repo = "cli";
+    rev = "v${version}";
+    hash = "sha256-rNWGd+yUDIrL2P+1XEce+HH8KDfThrTPXW+EOrV/n3o=";
```

Comment:
```suggestion
    hash = "sha256-rNWGd+yUDIrL2P+1XEce+HH8KDfThrTPXW+EOrV/n3o=";
    # A shortened git commit hash is part of the version output, and is
    # needd at build time. Use the `.git` directory to retrieve the 
    # commit SHA, and remove the directory afterwards since it is not needed
    # after that.
    leaveDotGit = true;
    postFetch = ''
      cd "$out"
      git rev-parse HEAD | cut --characters=1-7 | > "$out/COMMIT"
      find "$out" -name .git -print0 | xargs -0 rm -rf
    '';
```

Reply:
Ooh thanks, didn't know this was a thing / didn't think about it!